### PR TITLE
Fix: Handle git-cinnabar use cases

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Context in: https://bugzilla.mozilla.org/show_bug.cgi?id=1789814#c6

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Change Log
 
+## [3.1.1] - TBD
```
Comment (ahal) [inline CHANGELOG.md:3]: Since this release will include a `feat`, the next version should be `3.2.0`

File: src/taskgraph/util/vcs.py [hunk 2]
```diff
@@ -172,15 +198,10 @@ def branch(self):
...
-        if len(remotes) == 1:
-            return remotes[0]
-
-        if "default" in remotes:
-            return "default"
-
-        raise RuntimeError(
-            f"Cannot determine remote repository name. Candidate remotes: {remotes}"
+        return self._get_most_suitable_remote(
+            ("paths", "--quiet"),
```
Comment (ahal) [inline src/taskgraph/util/vcs.py:202]: What about if instead of passing in a `remote_command`, each vcs has:
```
@property
def remotes():
```
?

Comment (ahal) [reply src/taskgraph/util/vcs.py:202]: Similarly I wonder if `default_remote_name` should A) be a list of remote names, and B) be an attribute on the class.

**Review by ahal — CHANGES_REQUESTED** [review]
Thanks, looks good! Just a few issues / suggestions.

File: CHANGELOG.md (hunk 1 above)
Comment (JohanLorenzo) [reply CHANGELOG.md:3]: You're right! Done.

File: src/taskgraph/util/vcs.py (hunk 2 above)
Comment (JohanLorenzo) [reply src/taskgraph/util/vcs.py:202]: These are excellent points! I didn't like what I made with `remote_command`. I find your suggestion much more elegant! 💯 

Done

**Review by ahal — APPROVED** [review]
Thanks, looks great!

**Pull request merged**